components/script/dom/formdata.rs

```diff
@@ -70,20 +76,70 @@ impl FormData {
 
 impl FormDataMethods<crate::DomTypeHolder> for FormData {
     // https://xhr.spec.whatwg.org/#dom-formdata
-    fn Constructor(
+    fn Constructor<'a>(
         global: &GlobalScope,
         proto: Option<HandleObject>,
         can_gc: CanGc,
-        form: Option<&HTMLFormElement>,
+        form: Option<&'a HTMLFormElement>,
+        submitter: Option<&'a HTMLElement>,
     ) -> Fallible<DomRoot<FormData>> {
+        // Helper to validate the submitter
+        fn validate_submitter<'b>(
+            submitter: &'b HTMLElement,
+            form: &'b HTMLFormElement,
+        ) -> Result<FormSubmitterElement<'b>, Error> {
+            let error_not_a_submit_button =
+                Err(Error::Type("submitter is not a submit button".to_string()));
+
+            let element = match submitter.upcast::<Node>().type_id() {
```

Comment:
Why not:
```rust
submitter.downcast::<HTMLButtonElement>()
    .map(FormSubmitterElement::Button)
    .or_else(|| submitter.downcast::<HTMLInputElement>()
        .map(FormSubmitterElement::Input)
    )
```

Reply:
Thank you! Updated and rebased with a new [Try](https://github.com/shanehandley/servo/actions/runs/12924561967)